subsys/bluetooth/shell/ias.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file
+ * @brief Shell APIs for Bluetooth IAS
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <zephyr/types.h>
+#include <bluetooth/conn.h>
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/services/ias.h>
+#include "bt.h"
+
+extern const struct shell *ctx_shell;
+
+static void alert_stop(void)
+{
+	shell_print(ctx_shell, "Alert stopped\n");
+}
+
+static void alert_start(void)
+{
+	shell_print(ctx_shell, "Mild alert started\n");
+}
+
+static void alert_high_start(void)
+{
+	shell_print(ctx_shell, "High alert started\n");
+}
+
+static int cmd_ias_init(const struct shell *sh, size_t argc, char **argv)
+{
+	if (!ctx_shell) {
+		ctx_shell = sh;
+	}
+
+	BT_IAS_CB_DEFINE(ias_callbacks) = {
+		.no_alert = alert_stop,
+		.mild_alert = alert_start,
+		.high_alert = alert_high_start,
+	};
```

Comment:
These callbacks are registered before we call this function, aren't they?

Reply:
Yes, my mistake, now they are outside of init function. Furthermore, I think that init is no longer necessary, so I've removed it.